retrofit-adapters/rxjava/src/main/java/retrofit/ObservableExecutionAdapter.java

```diff
@@ -0,0 +1,97 @@
+package retrofit;
+
+import java.io.IOException;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import rx.Observable;
+import rx.Subscriber;
+import rx.functions.Action0;
+import rx.subscriptions.Subscriptions;
+
+/**
+ * TODO docs
+ */
+public final class ObservableExecutionAdapter implements CallAdapter {
+  /**
+   * TODO
+   */
+  private static ObservableExecutionAdapter create() {
+    return new ObservableExecutionAdapter();
+  }
+
+  private ObservableExecutionAdapter() {
+  }
+
+  @Override public Type parseType(Type returnType) {
+    if (Types.getRawType(returnType) != Observable.class) {
+      return null;
+    }
+    if (returnType instanceof ParameterizedType) {
+      return Types.getParameterUpperBound((ParameterizedType) returnType);
+    }
+    throw new IllegalStateException("Observable return type must be parameterized"
+        + " as Observable<Foo> or Observable<? extends Foo>");
+  }
+
+  @Override public <T> Observable<?> adapt(final Call<T> call, final Packaging packaging) {
```

Comment:
Here's a silly idea . . . you could your anonymous observable could be available in `Response` packaging only, and then you could use Rx to transform that to other packagings as necessary.